packages/hdwallet-native/src/native.ts

```diff
@@ -7,10 +9,57 @@ import { MixinNativeBTCWallet, MixinNativeBTCWalletInfo } from "./bitcoin";
 import { MixinNativeETHWalletInfo, MixinNativeETHWallet } from "./ethereum";
 import { MixinNativeCosmosWalletInfo, MixinNativeCosmosWallet } from "./cosmos";
 import { MixinNativeBinanceWalletInfo, MixinNativeBinanceWallet } from "./binance";
+import type { NativeAdapterArgs } from "./adapter";
+
+export enum NativeEvents {
+  MNEMONIC_REQUIRED = "MNEMONIC_REQUIRED",
+  READY = "READY",
+}
+
+export class NativeHDWalletBase {
+  readonly #events: EventEmitter2;
+
+  constructor() {
+    this.#events = new EventEmitter2();
+  }
+
+  get events() {
+    return this.#events;
+  }
+
+  /**
+   * Wrap a function call that needs a mnemonic seed
+   * @param hasMnemonic - A
+   * @param callback
```

Comment:
If we are going to doc comment the params, can we type it and provide a useful description if applicable?
`@param {boolean} hasMnemonic - useful description`

Reply:
I didn't meant to leave that in.